plugins/wdc/wdc-nvme.c

```diff
@@ -1038,6 +1039,84 @@ static int wdc_create_log_file(char *file, __u8 *drive_log_data,
 	return 0;
 }
 
+static int wdc_get_log(int fd, __u32 nsid, __u8 log_id, bool rae, __u8 uuid_ix,
```

Comment:
This looks like a rather generic routine, and very similar to existing common code. I think the common case only misses the uuid index. Any chance we can move this routine to the common case? I don't want every vendor duplicating this.

Reply:
Yes we could move it to nvme-ioctl.c and give it a generic name like nvme_get_log_from_uuid.  The other option would be to change the existing nvme_get_log routine to take a uuid parameter and call nvme_get_log14 as the routine I added does.  The problem with that option is that there are 28 calls to nvme_get_log so we would have to change a lot of code to implement that way.  A third option would be to remove the new routine and call nvme_get_log14 directly from get_dev_mgment_cbs_data. Please let me know your thoughts.